transport_analysis/analysis/velocityautocorr.py

```diff
@@ -5,111 +5,190 @@
-        Universe or group of atoms to apply this analysis to.
-        If a trajectory is associated with the atoms,
-        then the computation iterates over the trajectory.
-    select: str
-        Selection string for atoms to extract from the input Universe or
-        AtomGroup
+    atomgroup : AtomGroup
+        An MDAnalysis :class:`~MDAnalysis.core.groups.AtomGroup`.
+        Note that :class:`UpdatingAtomGroup` instances are not accepted.
+    dim_type : {'xyz', 'xy', 'yz', 'xz', 'x', 'y', 'z'}
+        Desired dimensions to be included in the VACF. Defaults to 'xyz'.
+    fft : bool
+        If ``True``, uses a fast FFT based algorithm for computation of
+        the VACF. Otherwise, use the simple "windowed" algorithm.
+        The tidynamics package is required for `fft=True`.
+        Defaults to ``True``.
 
     Attributes
     ----------
-    universe: :class:`~MDAnalysis.core.universe.Universe`
-        The universe to which this analysis is applied
-    atomgroup: :class:`~MDAnalysis.core.groups.AtomGroup`
+    atomgroup : :class:`~MDAnalysis.core.groups.AtomGroup`
         The atoms to which this analysis is applied
-    results: :class:`~MDAnalysis.analysis.base.Results`
-        results of calculation are stored here, after calling
-        :meth:`VelocityAutocorr.run`
-    start: Optional[int]
-        The first frame of the trajectory used to compute the analysis
-    stop: Optional[int]
-        The frame to stop at for the analysis
-    step: Optional[int]
-        Number of frames to skip between each analyzed frame
-    n_frames: int
-        Number of frames analysed in the trajectory
-    times: numpy.ndarray
-        array of Timestep times. Only exists after calling
-        :meth:`VelocityAutocorr.run`
-    frames: numpy.ndarray
-        array of Timestep frame indices. Only exists after calling
-        :meth:`VelocityAutocorr.run`
+    dim_fac : int
+        Dimensionality :math:`d` of the VACF.
+    results.timeseries : :class:`numpy.ndarray`
+        The averaged VACF over all the particles with respect to lag-time.
+        Obtained after calling :meth:`VelocityAutocorr.run`
+    results.vacf_by_particle : :class:`numpy.ndarray`
+        The VACF of each individual particle with respect to lag-time.
+    start : Optional[int]
+        The first frame of the trajectory used to compute the analysis.
+    stop : Optional[int]
+        The frame to stop at for the analysis.
+    step : Optional[int]
+        Number of frames to skip between each analyzed frame.
+    n_frames : int
+        Number of frames analysed in the trajectory.
+    n_particles : int
+        Number of particles VACF was calculated over.
     """
 
     def __init__(
         self,
-        universe_or_atomgroup: Union["Universe", "AtomGroup"],
-        select: str = "all",
-        # TODO: add your own parameters here
+        atomgroup: "AtomGroup",
```

Comment:
Does the `AtomGroup` type hint not work here once you have imported it?

Reply:
I noted this on Discord already but I just wanted to add it here so that the discussion is a bit more public. I tried to use mypy to check whether the `AtomGroup` type hint works, but I ran into some issues:
```
error: Need type annotation for "LONG_VERSION_PY" (hint: "LONG_VERSION_PY: Dict[<type>, <type>] = ...")  [var-annotated]
velocityautocorr.py:10: error: Cannot find implementation or library stub for module named "MDAnalysis.analysis.base"  [import]
velocityautocorr.py:10: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
velocityautocorr.py:154: error: Skipping analyzing "tidynamics": module is installed, but missing library stubs or py.typed marker  [import]
Found 6 errors in 2 files (checked 1 source file)
```
I looked this up for a while but I got stuck. Will be leaving it for the time being to work on the other items, but will revisit it.

Reply:
tbh mypy is kinda cursed. Not sure there is anything much better, but be aware of the `# type: ignore` comment if you ever feel like you are wasting too much time with it.

Reply:
You're right - I revisited it today and it has been too much trouble for something so minor. I went through a bunch of documentation, discussions, and even ChatGPT but had no luck. None of the errors mention the `AtomGroup` part and I wrote that part based on the cookiecutter template (only taking out the `Universe` part because it was unnecessary), so I think it is probably okay. I also ran it with the `# type: ignore` comment you mentioned. I think, at this point, I have done what I can.

Reply:
Sounds good.